e2e-tests/README.md

```diff
@@ -0,0 +1,151 @@
+requires decent access rights to make a push.
+
+To specify your own repository for the Operator docker image, you can use IMAGE environment variable:
+
+```
+export IMAGE=bob/my_repository_for_test_images:K8SPS-129-fix-feature-X
+```
+We use linux/amd64 platform by default. To specify another platform, you can use DOCKER_DEFAULT_PLATFORM environment variable
+
+```
+export DOCKER_DEFAULT_PLATFORM=linux/amd64
+```
+
+Use the following script to build the image:
+
+```
+./e2e-tests/build
+```
+
+Running all tests at once can be done with the following command:
+
+```
+kubectl kuttl test --config e2e-tests/kuttl.yaml --test "^test-name\$"
+```
+
+(see how to configure the testing infrastructure [here](#using-environment-variables-to-customize-the-testing-process)).
+
+Tests can also be run one-by-one as follows:
+
+```
+kubectl kuttl test --config e2e-tests/kuttl.yaml --test "^test-name\$"
+```
+
+Test names can be found in subdirectories of `e2e-tests/tests` (their names should be self-explanatory):
+
+```
+./e2e-tests/tests/async-ignore-annotations
+./e2e-tests/tests/gr-demand-backup
+./e2e-tests/tests/gr-one-pod
+./e2e-tests/tests/gr-users
+./e2e-tests/tests/one-pod
+./e2e-tests/tests/sidecars
+./e2e-tests/tests/auto-config
+./e2e-tests/tests/gr-demand-backup-haproxy
+./e2e-tests/tests/gr-recreate
+./e2e-tests/tests/haproxy
+./e2e-tests/tests/operator-self-healing
+./e2e-tests/tests/smart-update
+./e2e-tests/tests/config
+....
+```
+
+## Using environment variables to customize the testing process
+
+### Re-declaring default image names
+
+You can use environment variables to re-declare all default docker images used for testing. The
+full list of variables is the following one:
+
+* `IMAGE` - the Operator, `perconalab/percona-server-mysql-operator:main` by default,
+* `IMAGE_MYSQL` - Percona XtraDB Cluster, `perconalab/percona-server:main` by default,
+* `IMAGE_PMM_CLIENT` - Percona Monitoring and Management (PMM) client, `perconalab/pmm-client:dev-latest` by default,
+* `IMAGE_PROXY` - ProxySQL, `perconalab/percona-xtradb-cluster-operator:main-proxysql` by default,
+* `IMAGE_HAPROXY` - HA Proxy, `perconalab/haproxy:main` by default,
+* `IMAGE_BACKUP` - backups, `perconalab/percona-xtrabackup:main` by default,
+* `IMAGE_ORCHESTRATOR` - Orchestrator, `perconalab/percona-orchestrator:main` by default,
+* `IMAGE_ROUTER` - Router, `perconalab/percona-mysql-router:main` by default,
+* `IMAGE_TOOLKIT` - Percona Toolkit, `perconalab/percona-toolkit:main` by default,
+
+### Using automatic clean-up after testing
+
+By default, each test creates its own namespace and does not clean up objects in case of failure.
+
+To avoid manual deletion of such leftovers, you can run tests on a separate cluster and use the following environment variable to make the ultimate clean-up:
+
+```
+export CLEAN_NAMESPACE=1
+```
+
+**Note:** this will cause **deleting all namespaces** except default and system ones!
+
```

Comment:
you can maybe also add this:
`kubectl kuttl test --config e2e-tests/kuttl.yaml --test "^test-name\$" --skip-delete --skip-cluster-delete`
for if the user wants to investigate failures and so to skip the deletion.

Reply:
no we don't

Reply:
fixed :)